doc/developer/scan.txt

```diff
@@ -135,19 +60,252 @@ files, and section of the code they deal with, are :
 * ``scan_opt.py`` contains the list of all optimizations for the scan
```

Comment:
I would specify "Theano graph optimization". optimization is an overloaded word.